tensorflow_addons/callbacks/average_model_checkpoint.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import tensorflow as tf
+from tensorflow_addons.optimizers.average_wrapper import AveragedOptimizerWrapper
+
+
+class AverageModelCheckpoint(tf.keras.callbacks.ModelCheckpoint):
```

Comment:
NIT: add some comments here for the class. Otherwise LGTM :)

Reply:
> A question about AverageModelCheckpoint, when saving the moving average weights, we should not change the model.trainable_weights for we may keep training later. Does AverageModelCheckpoint change the model's weights after saving function?

@wuliytTaotao I'll use a parameter that allows both options:
1. Assign the moving average weights to the model, and save them.
2. Keep the old non-averaged weights, but the saved model uses the average weights. 

That will cover both scenarios of the user.